substrate/frame/revive/src/debug.rs

```diff
@@ -15,62 +15,155 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::{evm::TracerConfig, Config, DispatchError, Weight};
 pub use crate::{
+	evm::{CallLog, CallTrace, CallType, EthTraces, Traces},
 	exec::{ExecResult, ExportedFunction},
 	primitives::ExecReturnValue,
 };
-use crate::{Config, LOG_TARGET};
-use sp_core::H160;
+use alloc::{boxed::Box, format, string::ToString, vec::Vec};
+use sp_core::{H160, H256, U256};
 
 /// Umbrella trait for all interfaces that serves for debugging.
-pub trait Debugger<T: Config>: Tracing<T> + CallInterceptor<T> {}
+pub trait Debugger<T: Config>: CallInterceptor<T> {}
+impl<T: Config, V> Debugger<T> for V where V: CallInterceptor<T> {}
```

Comment:
I think we should unify those two things. Having the debugger as a type on the `Config` while `Tracer` is injected at runtime makes things confusing. Let's remove the `Debug` trait and its impl on `()`. It's function will move to the `Tracer` with a default impl that does nothing (as it is already the case).

We can also rename this module to `tracing`.

Reply:
Removed the callinterceptor for now.
Intercepting and mocking contracts can probably be done directly in the tooling rather than here in the runtime